app/components/inputFields/SelectOptionTextField.tsx

```diff
@@ -0,0 +1,61 @@
+import { MenuItem, TextField } from "@mui/material";
+import React, { FC } from "react";
+import USER_ROLE_OPTIONS from "@/app/constantData/userRoleOption";
+
+interface SelectOptionTextFieldProps {
+  defaultValue?: string;
+  label?: string;
```

Comment:
Use value and onChange Instead of defaultValue

defaultValue makes the component uncontrolled, which can lead to state desynchronization.
Instead, use a value prop with onChange for better control.
```suggestion
interface SelectOptionTextFieldProps {
  value: string;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  label?: string;
```

Reply:
made the changes and use it on component